https://gitlab.freedesktop.org/virgl/virglrenderer/-/merge_requests/1547
From: Dmitry Osipenko <dmitry.osipenko@collabora.com>
Date: Mon, 11 Aug 2025 13:17:56 +0300
Subject: [PATCH] drm: amdgpu: Fix clang warning about typeof() extension

AMDGPU nctx code uses typeof(), which is a GCC extension. Clang supports
typeof(), but emits a warning that breaks compilation because -Werror is
enabled by default. Silence the clang warning about typeof() like we do it
for drm_utils.

Closes: https://gitlab.freedesktop.org/virgl/virglrenderer/-/issues/627
Signed-off-by: Dmitry Osipenko <dmitry.osipenko@collabora.com>
Part-of: <https://gitlab.freedesktop.org/virgl/virglrenderer/-/merge_requests/1547>
--- a/src/drm/amdgpu/amdgpu_renderer.c
+++ b/src/drm/amdgpu/amdgpu_renderer.c
@@ -51,6 +51,7 @@
 
 #if defined(__clang__)
 #pragma GCC diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+#pragma GCC diagnostic ignored "-Wlanguage-extension-token"
 #endif
 
 #if 0
-- 
GitLab

